src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1315,6 +1317,25 @@ public String getPackageName() {
     @IntrinsicCandidate
     public native int getModifiers();
 
+    /**
+     * {@return an unmodifiable set of the {@linkplain AccessFlag access
+     * flags} for this class}
+     * @see #getModifiers()
+     * @jvms 4.1 The ClassFile Structure
+     * @jvms 4.7.6 The InnerClasses Attribute
+     * @since 19
+     */
+    public Set<AccessFlag> accessFlags() {
+        // This likely needs some refinement. Exploration of hidden
+        // classes, array classes.  Location.CLASS allows SUPER and
+        // AccessFlag.MODULE which INNER_CLASS forbids. INNER_CLASS
+        // allows PRIVATE, PROTECTED, and STATIC, which are not
+        // allowed on Location.CLASS.
+        return AccessFlag.maskToAccessFlags(getModifiers(),
```

Comment:
For this phase of the work, I was trying to avoid premature optimization of building caches, etc. Such refinement should certainly be considered later on.